modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRendererUtilities.java

```diff
@@ -545,6 +545,33 @@ public static List<GridCoverage2D> forceToValidBounds(
         return coverages;
     }
 
+    /**
+     * This method is checking if the 2 CRSs are equivalent by first using the equalsIgnoreMetadata
+     * approach and then looking for an identity transformation between the 2 CRSs if the first
+     * check fails.
+     *
+     * @param source The source {@link CoordinateReferenceSystem}
+     * @param target the target {@link CoordinateReferenceSystem}
+     */
+    public static boolean isEquivalentCRS(
+            CoordinateReferenceSystem source, CoordinateReferenceSystem target) {
+        boolean equivalentCRS = CRS.equalsIgnoreMetadata(source, target);
+        if (!equivalentCRS) {
+
+            try {
+                equivalentCRS = !CRS.isTransformationRequired(source, target);
+            } catch (FactoryException e) {
+                // That was an attempt looking for an identity transform.
+                // Let's ignore the exception and return the previous result
+                LOGGER.log(
+                        Level.FINE,
+                        "Unable to find MathTransform between sourceCrs and targetCrs",
```

Comment:
```suggestion
                        "Unable to find MathTransform between source and target CRS",
```

sourceCrs and targetCrs are no more.